custom_components/hacs/helpers/functions/misc.py

```diff
@@ -26,6 +27,7 @@ def get_repository_name(repository) -> str:
     )
 
 
+@lru_cache
```

Comment:
```suggestion
@lru_cache(maxsize=None)
```
maybe?

Reply:
They are almost all the same so the default was fine when I re-profiled it.

Reply:
It fails for py3.7 when not defined <https://github.com/hacs/integration/pull/1624/checks?check_run_id=1359323593#step:5:36>

Reply:
ahh. yes 3.7 👍

Reply:
Have not looked much in to lru, what happens when you reach the max?
Does it delete the oldes, ignores cache or fails?

Reply:
I did 1024 just to be on the safe side.... although I doubt someone would run home assistant that long that it would see enough repo changes that it actually mattered. 🤷

Reply:
To you and me that would seem farfetched, but it would not surprise me at all :smile: